### PR TITLE
Add versioning information for functions in InterfaceGenerator script

### LINKED ISSUE #3466: SDL doesn't reject the RPC from mobile in case the version of the mobile app is less then RPC version
### Bug Report
SDL doesn't reject the RPC from mobile in case the version of the mobile app is less then RPC version

##### Preconditions
1. HMI and SDL are started
2. Mobile app is registered with `syncMsgVersion = (minorVersion = 0, majorVersion = 5)`
3. Mobile app is activated

##### Reproduction Steps
1. App requests `CloseApplication`(RPC is available since=6.0)

##### Expected Behavior
SDL rejects the not supported RPC with INVALID_DATA result code

##### Observed Behavior
SDL processes the RPC successfully

##### OS & Version Information
* OS/Version: Ubuntu 18.04
* SDL Core Version: [develop](https://github.com/smartdevicelink/sdl_core/tree/develop)
* Testing Against: 
ATF: [develop](https://github.com/smartdevicelink/sdl_core/tree/develop)
Scripts: https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2438

### PR DESCRIPTION
**Pull request by jacobkeeler**
Fixes #3466

This PR is **ready** for review.

### Risk
This PR makes **no** API changes.

### Testing Plan
Follow steps in attached issue

### Summary
Adds versioning information for functions in InterfaceGenerator script and uses this to reject messages from mobile which aren't available in the negotiated RPC version.

##### Bug Fixes
* Reject messages from mobile which aren't available in the negotiated RPC version.

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/components/smart_objects/src/object_schema_item.cc [hunk 1]
```diff
@@ -132,6 +133,14 @@ errors::eType CObjectSchemaItem::validate(
         std::string validation_info = "Missing mandatory parameter: " + key;
         report->set_validation_info(validation_info);
         return errors::MISSING_MANDATORY_PARAMETER;
+      } else if (key.compare(msg_params) == 0) {
+        // If the message params struct was filtered, that means that the
+        // app's version is too low to use the message.
+        std::string validation_info =
```
Comment (JackLivio) [inline src/components/smart_objects/src/object_schema_item.cc:139]: SubscribeWayPoints is an example of an RPC with no params. Will this block SubscribeWayPoints?

Comment (JackLivio) [reply src/components/smart_objects/src/object_schema_item.cc:139]: I tested SubscribeWayPoints and it was successful

**Comment by JackLivio** [comment]
@jacobkeeler Can you please add an atf test case for function versioning?

File: src/components/smart_objects/src/object_schema_item.cc (hunk 1 above)
Comment (jacobkeeler) [reply src/components/smart_objects/src/object_schema_item.cc:139]: Yeah, `msg_params` should still exist in that case, it would just be empty

**Pull request merged**